guidelines.rst

```diff
@@ -154,6 +154,134 @@ The more stable the tier is, the stricter the rules for it are:
     - https://github.com/capi-workgroup/api-evolution/issues/42 (for limited API)
 
 
+One header
+==========
+
+All public API should be available after including ``Python.h``.
+
+To allow selecting an alternate API, such as a subset,
+use feature flags/macros that users define before including the header
+(e.g. ``Py_LIMITED_API``).
+Adding such a feature macro needs approval from the C API working group.
+
+.. note::
+
+    For background and discussions, see:
+
+    - https://github.com/capi-workgroup/api-evolution/issues/34
+
+
+.. XXX add a PEP number prefix to the anchor:
+
+.. _naming:
+
+Naming
+======
+
+[TODO: PEP 7 should be updated to link here once this PEP goes live.]
+
+All public names must be prefixed with ``Py``.
+
+Names that users should not use directly, but need to be visible to the
+compiler/linker, should be prefixed with ``_Py``.
+(Such names are not considered public API, that is, they should not appear in
+third-party source code.)
+
+This applies to all names in a global namespace: functions, macros, variables,
+typedefs, structs, enums, etc.; not to parameters or struct fields.
+
+The ``Py_`` prefix is reserved for global service routines like
+``Py_FatalError``; specific groups of APIs use a longer prefix,
+e.g. ``PyUnicode_`` for string functions.
+Use an existing prefix when applicable. If you want to add a new prefix,
+contact the C API working group.
+
+The ``Py_`` prefix is in mixed case, even in macro names.
+For example: ``PyUnicode_AS_STRING``.
+(Several existing macros use the upper-case ``PY``; if you need this prefix
+for consistency, please get approval from the C API working group.)
+
+Unstable API is prefixed with ``PyUnstable_`` instead of ``Py``,
+for example ``PyUnstable_Long_IsCompact`` or (hypothetically)
+``PyUnstable_String_GET_SIZE``.
+
+.. note::
+
+    For background and discussions, see:
+
+    - https://github.com/capi-workgroup/api-evolution/issues/21
+
+
+Do not reuse names
+------------------
+
+If an API's behavior changes in a backwards-incompatible way,
+for example when a function's signature changes,
+add new API with a new name.
+You can deprecate and remove the old version following Python's
+:pep:`backwards compatibility policy <387>`.
+
+After API has been removed, do not reuse the old name,
```

Comment:
It could be helpful to have a list of "retired" names to check for conflicts.

Reply:
Yes. On the other hand,
- I don't think it's too likely that someone reuses a name accidentally, and
- if they do, it's a big problem.

The main point is that you shouldn't intentionally retire API to make its name available.